datascience/maps.py

```diff
@@ -0,0 +1,510 @@
+"""
+
+Integrates the folium package to draw maps.
+
+"""
+
+from IPython.core.display import HTML
+import json
+import folium
+import random
+
+#########################
+# Exposed Functionality #
+#########################
+
+
+def draw_map(center=None, **kwargs):
+    """ 
+    Draw a map with center & zoom containing all points and
+    regions that displays points as circles and regions as polygons.
+    """
+
+    return Map(
+        location=center,
+        **kwargs).map().to_html()
+
+
+#########
+# Utils #
+#########
+
+
+def _map_to_defaults(defaults, params, excludes=[], includes=[]):
+    """
+    Squishes together two dictionaries, with one containing defaults
+    """
+    rval = defaults.copy()
+    rval.update(params)
+    return {
+        k: v for k, v in rval.items() 
+        if k not in excludes
+        and k in (includes or rval.keys())
+    }
+
+
+def _hook(f):
+    """
+    Decorator for obj method, checks for PREFIX_before and PREFIX_after 
+    a method is called 
```

Comment:
nit: This doesn't just check for `PREFIX_before / after`, it calls 'em, so we can be more clear here.